internal/updater/updates.go

```diff
@@ -47,54 +49,45 @@ func findUpdatesSerial(ctx context.Context, cfg *PartialConfig, filename string)
 
 func findUpdatesParallel(ctx context.Context, cfg *PartialConfig, filename string) (*UpdateSet, error) {
 	numUpdates := len(cfg.Components)
-	jobChan := make(chan WorkerJob, numUpdates)
 	resChan := make(chan *ChangeSet, numUpdates)
 	errChan := make(chan error, numUpdates)
 
-	var wg sync.WaitGroup
-	var numWorkers = 4
-
-	// Start 4 workers
-	for i := 0; i < numWorkers; i++ {
-		wg.Add(1)
-		go func() {
-			defer wg.Done()
+	var numWorkers = int(math.Ceil(float64(runtime.NumCPU() / 2)))
```

Comment:
might be a good idea to have a minimum of 2 workers?

Reply:
Yeah, not opposed to that. Will update the worker calc to have 2 as minimum